### PR TITLE
ci: compile the project using GitHub actions

### PR DESCRIPTION
**Pull request by zanna-37**
With this new action the project is compiled every time a new release is published and the output file is attached to it. No need to compile manually and create a commit for it.
If you decide to use this I suggest you to delete the dist folder from the repo and add it to the `.gitignore`.

No changes for HACS, everything continue to work.

### CODE REVIEW

**Comment by dbuezas** [comment]
This is awesome!

**Comment by dbuezas** [comment]
I think this will also put a download counter in hacs.

**Comment by zanna-37** [comment]
> I think this will also put a download counter in hacs. 

Yes, a counter for the earliest release. ☺️

**Comment by dbuezas** [comment]
> Yes, a counter for the earliest release. ☺️

What do you mean with "the earliest" ?

**Comment by dbuezas** [comment]
BTW, my manual workflow was to call `npm version patch`(or minor or major) and an npm hook builds it. This is better, but it wasn't _that_ bad

**Comment by dbuezas** [comment]
> If you don't need it, feel free to reject the PR, no offense.

I don't _need_ this. I *want* it! :)

I wanted to do this for a long while, but never got broke the "have to learn first" barrier.

Where did you learn how to do this?

**Comment by zanna-37** [comment]
Same way I learnt ~90% of what I know. _I want it and I'm not giving up until it's done._ 😂

Jokes aside, github can be a wonderful place, people usually share a lot of problems and some solve them before others. It's just a matter of getting lost in projects and being curious.
And internet is full of good documentation and guides.

The only thing difficult is having free time, because "getting lost" requires a lot of it. 😂


**TL;DR**
I'm a very curious person, and I like to discover how people (more knowledgeable than me) do stuff and then take inspiration. 😉

**Comment by dbuezas** [comment]
Same as me. So copy & paste? 🤣

**Comment by zanna-37** [comment]
It my experience it rarely its that easy.
It's:
- copy
- paste
- stackoverflow
- search on github to see if someone is doing the same
- realizing that you'd better read the docs
- accept that you need to understand the pasted code
- doing a better version that fits your needs

It's always starts as a "well can't be that difficult" and ends in "I did so many mistakes that I'm now an expert" 😂
